DeviceBridgeE2E/src/device-bridge.spec.ts

```diff
@@ -200,114 +274,270 @@ test.serial('Test device to cloud messaging', async t => {
-        });
+        await got
+            .delete<any>(url(), { headers })
+            .then(async result => {
+                t.assert(
+                    result.statusCode == 200 || result.statusCode == 204,
+                    `Status code was ${
+                        result.statusCode
+                    } instead of 200 for url: ${url()}`
+                );
+            })
+            .catch((reason: any) => {
+                // Assert has a 400, 404 or 405 if failed
+                // 400 is okay as auth comes before 400 is triggered, we are just testing auth
+                t.assert(
+                    (reason.message as string).indexOf('404') > 0 ||
+                        (reason.message as string).indexOf('405') > 0 ||
+                        (reason.message as string).indexOf('400') > 0,
+                    reason.message
+                );
+            });
         // Test delete, no auth
-        await got.delete<any>(url()).then(async (result) => {
-            t.fail();
-        }).catch((reason: any) => {
-            // Assert has a 401 or 405 if failed
-            t.assert((reason.message as string).indexOf("401") > 0 || (reason.message as string).indexOf("405") > 0, reason.message)
-        });
+        await got
+            .delete<any>(url())
+            .then(async result => {
+                t.fail();
+            })
+            .catch((reason: any) => {
+                // Assert has a 401 or 405 if failed
+                t.assert(
+                    (reason.message as string).indexOf('401') > 0 ||
+                        (reason.message as string).indexOf('405') > 0,
+                    reason.message
+                );
+            });
+    }
+});
+
+test.serial('Test restart', async t => {
+    // Create subscription to Azure Function
+    const callbackUrl = `${t.context.ctx.callbackUrl}&deviceId=${t.context.device.id}`;
+    await t.context.ctx.deviceBridgAPI.createConnectionStatusSubscription(
+        t,
+        t.context.device.id,
+        callbackUrl
+    );
+    await sleep(3000);
+    // Ensure get works
+    var response = await t.context.ctx.deviceBridgAPI.getConnectionStatusSubscription(
+        t,
+        t.context.device.id
+    );
+    t.is(callbackUrl, response.body.callbackUrl);
+    await t.context.ctx.deviceBridgAPI.createCMDSubscription(
+        t,
+        t.context.device.id,
+        callbackUrl
+    );
+    await sleep(3000);
+    // Delete echo for conection status
+    await t.context.ctx.deviceBridgAPI.deleteEcho(t, t.context.device.id);
+    // Restart container
+    await got.post<{ [name: string]: string }>(t.context.ctx.restartApiUrl, {
+        json: {
+            request: {},
+        },
+        responseType: 'json',
+        headers: {
+            Authorization: 'Bearer ' + t.context.ctx.restartBearerToken,
+        },
+    });
+
+    var attempts = 24;
+    for (var i = 0; i < attempts; i++) {
```

Comment:
The test should fail if the attempts are exhausted but the event was never received (currently it will finish successfully if none of the retries succeeded).